File: IKeep.Lib/Models/Area.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IKeep.Lib.Models
{
    public class Area : Entity
    {
        public string Ref { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Guid FloorId { get; set; }
        [JsonIgnore]
        public virtual Floor Floor { get; set; }
        [JsonIgnore]
        public virtual ICollection<Element> Elements { get; set; }
        public List<Guid> ElementsIds
        {
            get
            {
                return Elements == null ? new List<Guid>() : Elements.Select(x => x.Id).ToList();
            }
        }
        //¿Es necesario?
        public virtual Map Map { get; set; }
    }
}

File: IKeep.Web/ClientApp/src/models/Area.js
import Entity from './Entity'

class Area extends Entity
{
    constructor(json)
    {
        super(json)
        if(json)
        {
            this.Ref = json.ref;
            this.Name = json.name;
            this.Description = json.description;
            this.FloorId = json.floorId;
            this.ElementsIds = json.elementsIds;
            this.Map = json.map;
        }
        else
        {
            this.Ref = "";
            this.Name = "";
            this.Description = "";
            this.FloorId = "00000000-0000-0000-0000-000000000000";
            this.ElementsIds = [];
            this.Map = "";
        }
        
    }
}

export default Area;